File: 백준/Bronze/1157. 단어 공부/단어 공부.c
#include <stdio.h>
#include <string.h>

int main(void) {
	int alpha[26] = { 0, };
	char arr[1000001];
	scanf("%s", arr);
	for (int i = 'a'; i <= 'z'; i++) {
		for (int j = 0; j < strlen(arr); j++) {
			if (arr[j] == i) {
				alpha[i - 'a']++;
			}
		}
	}
	for (int i = 'A'; i <= 'Z'; i++) {
		for (int j = 0; j < strlen(arr); j++) {
			if (arr[j] == i) {
				alpha[i - 'A']++;
			}
		}
	}

	int max = alpha[0];
	int number = 0;
	for (int i = 1; i < 26; i++) {
		if (max < alpha[i]) {
			max = alpha[i];
			number = i;
		}
	}
	int cnt = 0;
	for (int i = 0; i < 26; i++) {
		if (max == alpha[i]) {
			cnt++;
		}
	}

	if (cnt > 1) {
		printf("?\n");
	}
	else {
		printf("%c\n", number + 'A');
	}
	return 0; 
}

File: 백준/Bronze/1157. 단어 공부/단어 공부.py
arr = input()
alpha = [0] * 26

for char in arr:
    if(char >= 'a' and char <= 'z'):
        alpha[ord(char) - ord('a')] += 1
    if(char >= 'A' and char <= 'Z'):
        alpha[ord(char) - ord('A')] += 1
            
max = alpha[0]
cnt = 0
number = 0
for i in range(26):
    if (max < alpha[i]):
        max = alpha[i]
        number = i
for i in alpha:
    if (max == i):
        cnt += 1
if(cnt > 1):
    print("?")
else:
    print(chr(number + 65))